packages/blade/src/components/ButtonGroup/_decisions/decisions.md

```diff
@@ -0,0 +1,138 @@
+
+This document outlines the API details of the `ButtonGroup` component, encompassing their structural composition, functional attributes, and visual representation.
+
+<img width="100%" src="./button-group-variants.png" alt="Button Group Variants">
+
+- [Design](#design)
+- [`ButtonGroup` API](#buttongroup-api)
+- [Usage](#usage)
+- [Accessibility](#accessibility)
+
+## Design
+
+[Figma Link](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade-DSL?type=design&node-id=80700%3A197616&mode=design&t=7DAMvXAfVzPVN7Kl-1) to all variants of the `ButtonGroup` component.
+
+## `ButtonGroup` API
+
+```ts
+type ButtonGroupProps = {
+  /**
+   * Accepts multiple Button components as children
+   */
+  children: React.ReactNode;
+  /**
+   * Specifies the visual style variant of the ButtonGroup.
+   *
+   * @default 'primary'
+   */
+  variant?: ButtonProps['variant'];
+  /**
+   * Specifies the size of the ButtonGroup.
+   *
+   * @default 'medium'
+   */
+  size?: ButtonProps['size'];
+  /**
+   * Specifies the color of the ButtonGroup.
+   *
+   * @default 'primary'
+   */
+  color?: ButtonProps['color'];
+  /**
+   * Specifies whether the ButtonGroup should take up the full width of its container.
+   */
+  isFullWidth?: boolean;
+  /**
+   * Disables or enables the ButtonGroup component
+   *
+   * @default false
+   */
+  isDisabled?: boolean;
+  /**
+   * Test ID for automation
+   */
+  testID?: string;
+};
+```
+
+## Usage
+
+- Use button groups to visually associate multiple buttons that share a common purpose or functionality.
+
+  <img width="50%"  src="./button-group-usage.png" alt="Button Group Usage" />
+
+  ```tsx
+  import {
+    ButtonGroup,
+    Button,
+    RefreshIcon,
+    ShareIcon,
+    DownloadIcon,
+  } from '@razorpay/blade/components';
+
+  const App = () => {
+    return (
+      <ButtonGroup>
+        <Button icon={RefreshIcon} />
+        <Button icon={ShareIcon}>Share</Button>
+        <Button icon={DownloadIcon}>Download</Button>
+      </ButtonGroup>
+    );
+  };
+  ```
+
+- Buttons inside a `ButtonGroup` can be used to trigger a dropdown.
+
+  > [!IMPORTANT]
+  >
+  > We will introduce a new `placement?:| "bottom-end" | "bottom-start" | "left-end" | "left-start" | "right-end" | "right-start" | "top-end" | "top-start"` prop in the `DropdownOverlay` component to control the placement of the dropdown.
+
+  <img width="100%" src="./button-group-dropdown-usage.png" alt="Button Group Usage with Dropdown" />
+
+  ```tsx
+  import {
+    ButtonGroup,
+    Button,
+    Dropdown,
+    DropdownOverlay,
+    DropdownButton,
+    ActionList,
+    ActionListItem,
+    ChevronDownIcon,
+  } from '@razorpay/blade/components';
+
+  const App = () => {
+    return (
+      <ButtonGroup>
+        <Button icon={PlusIcon}>Payout</Button>
+        <Dropdown>
```

Comment:
So button group will allow Button & Dropdown components? What if someone uses  some weird dropdown trigger like SelectInput? Are we also adding a check that will restrict only using `DropdownButton` inside `Dropdown`?

Reply:
>So button group will allow Button & Dropdown components? What if someone uses some weird dropdown trigger like SelectInput?

Consumers control the children's content, however, they won't pass any random thing unless it comes from the design. 

>Are we also adding a check that will restrict only using DropdownButton inside Dropdown?

We can add checks, but it may increase the complexity because `DropdownButton`  is nested inside `Dropdown`. I'll need to explore the feasibility.

Reply:
We can add the check it would look something like below:
```js
      if (__DEV__) {
            // throw error if child is not a button or dropdown with button trigger
            if (
              child?.type?.componentId !== 'Button' &&
              !(
                child?.type?.componentId === 'Dropdown' &&
                child.props.children[0].type.componentId === 'DropdownButton'
              )
            ) {
              console.error(
                `ButtonGroup only accepts button or dropdown elements as children.
              );
            }
          }
```